crates/router/src/db/configs.rs

```diff
@@ -100,7 +94,7 @@ impl ConfigInterface for Store {
         key: &str,
         config_update: storage::ConfigUpdate,
     ) -> CustomResult<storage::Config, errors::StorageError> {
-        cache::publish_and_redact(self, CacheKind::Config(key.into()), || {
+        cache::db_call_with_redact_and_publish(self, [CacheKind::Config(key.into())], || {
             self.update_config_in_database(key, config_update)
         })
```

Comment:
Should we consider merging `update_config_in_database()` and this method? We can take that up as a separate PR, not a problem.